scalajslib/src/mill/scalajslib/ScalaJSModule.scala

```diff
@@ -73,4 +80,41 @@ trait ScalaJSModule extends scalalib.ScalaModule { outer =>
 
 }
 
-trait TestScalaJSModule extends ScalaJSModule with TestModule
\ No newline at end of file
+trait TestScalaJSModule extends ScalaJSModule with TestModule {
+  def scalaJSTestDeps = T {
+    Loose.Agg(
+      ivy"org.scala-js::scalajs-library:${scalaJSVersion()}",
+      ivy"org.scala-js::scalajs-test-interface:${scalaJSVersion()}"
+    )
+  }
+
+  def fastOptTest = T {
+    val bridge = scalaJSLinkerBridge(scalaJSLinkerClasspath().map(_.path))
+
+    val linkLibs = resolveDeps(scalaJSTestDeps)() ++ compileDepClasspath()
+
+    link(
+      main = None,
+      (compile() +: upstreamCompileOutput()).map(_.classes.path),
+      linkLibs.map(_.path),
+      bridge,
+      FastOpt
+    )
+  }
+
+  override def test(args: String*) = testLocal(args:_*)
```

Comment:
There is a strange dir structure produced here. It looks like this:
```
out/
├── milljs
│   ├── compile
│   │   ├── log
│   │   └── meta.json
│   └── test
│       └── testLocal
│           └── overriden
│               └── mill.scalajslib.TestScalaJSModule#testLocal
│                   ├── log
│                   └── meta.json
```

Reply:
Not sure it should be this way. in batch mode mill fails with exception: 
```
Exception in thread "main" java.io.FileNotFoundException: /Users/rockjam/projects/mill-scala-js/out/milljs/test/testLocal/meta.json
```

Reply:
I think wrapping this e.g.

```scala
override def test(args: String*) = T.command{ testLocal(args:_*) }
```

Should do the right thing?

Reply:
Well, I tried, and it became: 

```
out
├── milljs
│   ├── test
│   │   ├── test
│   │   │   └── overriden
│   │   │       └── mill.scalajslib.TestScalaJSModule#test
│   │   │           └── meta.json
│   │   ├── testLocal
│   │   │   └── overriden
│   │   │       └── mill.scalajslib.TestScalaJSModule#testLocal
│   │   │           ├── log
│   │   │           └── meta.json
```

Reply:
@lihaoyi What if don't extend `TestModule` here, and extend only `TaskModule` and implement `test` without override?

Reply:
For instance: https://github.com/lihaoyi/mill/pull/119/commits/777dfd33e0a06305bae749d5617304fc90d805a1